url4c41254592_topskychen_AuthTopk_tgz-pJ8-quickhull3d_QuickHull3DTestJ8/src/utility/security/RSA.java

```diff
@@ -0,0 +1,244 @@
+	 * 
+	 * @param keySize 
+	 */
+	
+	public void initKey(int keySize){
+		if(isLoad){
+			/*n = new BigInteger("131884943464045003582223572764312780891162693721585194592930524324036800924838681762453491291771217265809865991455743941676406325061202522717805990960640954746194204966663488362888261025109842355102675226877485643946372964765916251796647608347313211787097206099009857242384502137752539619309766105281035929537");
+			e = new BigInteger("162259276829213363391578010288127");
+			d = new BigInteger("83449415258234725851576926589709492036325417822780187671214268219104557317967121765323271466150154985252796121180460609454398453624143877649330539413762235154006540684170816796948035994114344229672746485878680815922649054243988474628827807343372872130363224420020644927099041494725797963189924214734565337323");
+			*/
+			/*BigInteger p = new BigInteger("7804947815828153757360229819652411239902581336788694465424613098446329573400278070343537820396291896016908012945254519909616381770357121621312154337477549");
+			BigInteger q = new BigInteger("12247127755728974923191139153732636477838906492686144861614769968099366746368758706323214837421927857281933445666383738321618845728660233681677187621034241");
+	    	*/
+			n = new BigInteger("9137102646213891968873960082305956179850467281348997924378425537822941327328353711486252262922644444840455727525901197729954054449881680414499250710131588148276328442397092515587006832309623836117942996692050118217219959552770621784470482084525859284361578517199520103365067471771138969454007315313886974448656193939941734279189230182934062065753837306411996014007893758334981553399496819364055499244835592090097892734327146277218349751641763468959032202918273753078465610458996364688468982697036369663280854824811270106731404956468032313803254120975168652977941430279563917186236894004867711790171585042367637685481");
+			//BigInteger m = (p.subtract(BigInteger.ONE))
+	        //         .multiply(q.subtract(BigInteger.ONE)).multiply(p).multiply(q);
+			e =	RSA.PRIME_P;
+		    d = new BigInteger("1134521025461381199780628738339629122047482384507794556518214256222766407939240641355301462595725727630403055841439610538167370181176024435785171895624139924670057634156010264713227319017584709542026521251989552610333061298960431812931715338334915441996250974446628953096609645461395560558237259379836497769063763957643894725229107010847869771811148741932466859705490123089946951666542994825250397188744462581617218567386450896987696984867870226835922454146094716321666838622444420976707886718463203408506658584309274617841897944797505918670232362861870371742229872090674318689314242776254164281194707978579029940223");
+			//while(m.gcd(e).intValue() > 1) e = e.add(new BigInteger("2"));
+		    //d = e.modInverse(m);
+		    //System.out.println(e);
+		    //System.out.println(d);
+			DataInputStream dis = null;
+			if(DEBUG){
+				try {
+					dis = new DataInputStream(new FileInputStream("RSA.seeds"));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *dis*. Execution paths that do not contain closure statements were detected. Ensure *dis* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).